library/ssl_misc.h

```diff
@@ -1037,6 +1037,13 @@ void mbedtls_ssl_set_inbound_transform( mbedtls_ssl_context *ssl,
 void mbedtls_ssl_set_outbound_transform( mbedtls_ssl_context *ssl,
                                          mbedtls_ssl_transform *transform );
 
+#if defined(MBEDTLS_SSL_SERVER_NAME_INDICATION)
```

Comment:
`MBEDTLS_SSL_SERVER_NAME_INDICATION` should be added into `mbedtls_config.h`. 

- add `TLS1_3` prefix.
- add dependency check in  `check_config.h`